packages/mask/src/plugins/Wallet/services/chain.ts

```diff
@@ -1,117 +1,77 @@
-import { throttle } from 'lodash-unified'
-import { BalanceOfChains, ProviderType } from '@masknet/web3-shared-evm'
-import { pollingTask } from '@masknet/shared-base'
+import { debounce, DebouncedFunc, noop } from 'lodash-unified'
+import { EthereumAddress } from 'wallet.ts'
+import { getEnumAsArray } from '@dimensiondev/kit'
+import { ChainId, ProviderType } from '@masknet/web3-shared-evm'
 import { getBalance, getBlockNumber, resetAllNonce } from '../../../extension/background-script/EthereumService'
 import { startEffects } from '../../../../utils-pure'
-import { UPDATE_CHAIN_STATE_DELAY } from '../constants'
 import {
     currentMaskWalletAccountSettings,
     currentAccountSettings,
-    currentBalanceSettings,
-    currentBlockNumberSettings,
     currentChainIdSettings,
-    currentMaskWalletBalanceSettings,
     currentMaskWalletChainIdSettings,
     currentProviderSettings,
-    currentBalancesSettings,
+    currentBalanceOfChainSettings,
+    currentBlockNumberOfChainSettings,
 } from '../settings'
 
-let beats = 0
 const { run } = startEffects(import.meta.webpackHot)
 
-export async function kickToUpdateChainState() {
-    beats += 1
+async function updateBalanceOfChain(chainId: ChainId, address: string) {
+    if (!address || !EthereumAddress.isValid(address)) return
+    const balance = await getBalance(address, {
+        chainId,
+        providerType: ProviderType.MaskWallet,
+    })
+    currentBalanceOfChainSettings.value = {
+        ...currentBalanceOfChainSettings.value,
+        [address]: {
+            [chainId]: balance,
+        },
+    }
 }
 
-export async function updateBalances(data: BalanceOfChains) {
-    const balancesOfChains = { ...currentBalancesSettings.value }
-    for (const [key, value] of Object.entries(data)) {
-        balancesOfChains[key] = {
-            ...balancesOfChains[key],
-            ...value,
-        }
+async function updateBlockNumberOfChain(chainId: ChainId) {
+    const blockNumber = await getBlockNumber({
+        chainId,
+    })
+    currentBlockNumberOfChainSettings.value = {
+        ...currentBlockNumberOfChainSettings.value,
+        [chainId]: blockNumber,
     }
-
-    currentBalancesSettings.value = balancesOfChains
```

Comment:
If you are replacing `currentBalancesSettings` with `currentBlockNumberOfChainSettings`, please replace globally